scipy/optimize/_lbfgsb_py.py

```diff
@@ -279,24 +279,25 @@ def _minimize_lbfgsb(fun, x0, args=(), jac=None, bounds=None,
     x0 = asarray(x0).ravel()
     n, = x0.shape
 
+    # historically old-style bounds were/are expected by lbfgsb.
+    # That's still the case but we'll deal with new-style from here on,
+    # it's easier
     if bounds is None:
-        bounds = [(None, None)] * n
-    if len(bounds) != n:
-        raise ValueError('length of x0 != length of bounds')
-
-    # unbounded variables must use None, not +-inf, for optimizer to work properly
-    bounds = [(None if l == -np.inf else l, None if u == np.inf else u) for l, u in bounds]
-    # LBFGSB is sent 'old-style' bounds, 'new-style' bounds are required by
-    # approx_derivative and ScalarFunction
-    new_bounds = old_bound_to_new(bounds)
+        bounds = np.empty((2, n), dtype=float)
```

Comment:
Maybe we could avoid creating this array in the first place and only update `nbd` later on if if `bounds is not None`. Note that `nbd` is initialized with zeros so no action needed in that case.

Reply:
I've taken this statement on board. TBH, I don't think it's a major imposition to create such an array, it typically won't be that big of a memory hog, or take long to create.